Updated all icons to reference a single component [fixes #758]

carlfairclough opened the pull request:
Updated all icons to reference a single icon component

## Description

A single `<icon name="..." /> component can now be used to generate icons detailed in the style guide. These have been implemented in the header.

They accept the following props:
```
lightModeColor: String,
size: String, Number
```

`darkModeColor` has not been implemented as it is reliant on a class which exists higher in the DOM tree. To target dark mode, use the following in CSS:
`#wrapper.dark-mode .unique-class svg path { fill: pink }`

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/4670881/74899170-574bc680-5359-11ea-8b0b-39fe4e984f02.png)
![image](https://user-images.githubusercontent.com/4670881/74899193-6af72d00-5359-11ea-86b7-85be1da28616.png)
![image](https://user-images.githubusercontent.com/4670881/74899240-88c49200-5359-11ea-9154-4925307c8037.png)

samajammin commented on docs/.vuepress/components/Icon.vue:
```diff
@@ -0,0 +1,121 @@
+<template>
+  <svg
```
Where'd you get all these SVG files? Do you think this is the best approach vs. using an icon library, e.g. [Font Awesome](https://github.com/FortAwesome/vue-fontawesome)? This SVG approach feels like it could get a bit unwieldy if we add more icons down the line.

carlfairclough replied:
(Commenting for visibility)
These are directly from the figma file, they all share size & visual weight. It might be unwieldy in future I agree but imagine there will be minor updates now that the bulk of this is done (i.e. darkmode handling, passing colors, etc)

samajammin commented on docs/.vuepress/theme/components/Header.vue:
```diff
@@ -22,53 +22,61 @@
-          class="show-dark"
-          src="../images/icon-moon.svg"
-        />
+        <icon name="lightmode" class="hide-dark" />
```
Rather than having 2 DOM elements & additional classes (`hide-dark` & `show-dark`) to handle the display logic, I'd rather just pass `darkMode` property from the Layout component into Header, then update the icon `name` based on that to present the correct icon. Make sense? You can see an example in Layout where we pass that property to the Hero component.

carlfairclough replied:
Ah I'd missed that, I wasn't entirely sure how to handle it & I prefer what you suggest. Ty!

carlfairclough commented:
@samajammin I've updated the icon to handle isDarkMode, and I'm using `v-if` to display the correct icon vs hiding with CSS.

I also updated dark/darkmode props to be a type-checked isDarkMode prop (Bool). This means there have been some minor updates which reach into components like the Hero.

I've removed all unused icons, with the caveat that the list icons are now base64 encoded directly in the CSS, as it's likely those styles will removed/updated soon anyway.

samajammin commented on docs/.vuepress/components/Icon.vue:
```diff
@@ -0,0 +1,128 @@
+    },
+    darkModeColor: {
+      type: String,
+      default: '#FFF'
```
Instead of the hex color `#FFF`, I'd like to see us import variables from config.styl, e.g. `$colorWhite`

carlfairclough replied:
We can't pass a stylus variable into this but we *can* pass CSS variables. I'm happy to go with CSS vars, but it's a departure from the current system

samajammin commented on docs/.vuepress/components/Icon.vue:
```diff
@@ -0,0 +1,128 @@
+    },
+    lightModeColor: {
+      type: String,
+      default: '#000'
```
Same here.

samajammin commented on docs/.vuepress/theme/components/Header.vue:
```diff
@@ -22,53 +23,57 @@
 }
 </script>
 
+// unscoped css to pass styles to navigations
```
Yay comments!

samajammin commented on docs/.vuepress/theme/components/SearchBox.vue:
```diff
@@ -153,10 +154,24 @@ export default {
 @import '../styles/config.styl'
 
+#wrapper.dark-mode .search-box svg path
+  fill: $subduedColor
```
Nice use of variable 👍

samajammin commented on docs/.vuepress/theme/components/SidebarButton.vue:
```diff
@@ -1,44 +1,47 @@
+      <icon
+        name="menu"
+        :isDarkMode="isDarkMode"
+        lightModeColor="#000"
```
Same here - would prefer variables.

samajammin commented on docs/.vuepress/theme/styles/dark.styl:
```diff
@@ -35,17 +35,15 @@
           color $textColorDark
           &.highlight
-            background-image url(../images/highlight-dark.svg)
+            background-image url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iMjgwIiBoZWlnaHQ9IjI4IiB2ZXJzaW9uPSIxLjEiIHZpZXdCb3g9IjAgMCAyODAgMjgiPjx0aXRsZT5GaWxsIDE8L3RpdGxlPjxkZXNjPkNyZWF0ZWQgd2l0aCBTa2V0Y2guPC9kZXNjPjxnIGlkPSJQYWdlLTEiIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCIgc3Ryb2tlPSJub25 ...
```
Why are we doing this?
